src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java

```diff
@@ -176,4 +176,24 @@ void should_create_token_when_cache_is_expired() throws InterruptedException {
         verify(users, times(2)).getUser(any());
         verify(idamClient, times(2)).authenticateUser(any(), any());
     }
+
+    @Test
+    void should_create_new_token_when_token_removed_from_cache() {
+
+        String jurisdiction = "probate";
+
+        given(users.getUser(jurisdiction)).willReturn(new Credential(USERNAME, PASSWORD));
+        given(idamClient.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT, JWT2);
+        String token1 = idamCachedClient.getAccessToken(jurisdiction);
+
+        idamCachedClient.removeAccessTokenFromCache(jurisdiction);
+
+        String token2 = idamCachedClient.getAccessToken(jurisdiction);
+
+        assertThat(JWT).isEqualTo(token1);
+        assertThat(JWT2).isEqualTo(token2);
+        assertThat(token1).isNotEqualTo(token2);
```

Comment:
This assertion is probably unnecessary.

Reply:
I think we can keep it. may be other two is unnecessary but still we can keep all